src/main/java/net/masterthought/cucumber/generators/AbstractPage.java

```diff
@@ -32,30 +12,20 @@
  */
 public abstract class AbstractPage {
 
-    private static final Logger LOG = LogManager.getLogger(AbstractPage.class);
-
-    private final VelocityEngine engine = new VelocityEngine();
-    protected final VelocityContext context = new VelocityContext();
-
     /** Name of the HTML file which will be generated. */
     private final String templateFileName;
-    /** Results of the report. */
-    protected final ReportResult reportResult;
-    /** Configuration used for this report execution. */
-    protected final Configuration configuration;
-
-    protected AbstractPage(ReportResult reportResult, String templateFileName, Configuration configuration) {
+    
+    protected AbstractPage(String templateFileName) {
         this.templateFileName = templateFileName;
-        this.reportResult = reportResult;
-        this.configuration = configuration;
-
-        this.engine.init(buildProperties());
```

Comment:
A less intrusive way to reduce memory consumption that I've tried is to just use the singleton `VelocityEngine` API. So far I haven't noticed any negative impact with that change and it only affects the internals of AbstractPage without any other API changes.

Reply:
Velocity singleton model is based in static variables that affects the whole jvm. The use of static variables have a lot of problems, and usually they are the root of memory leaks in j2ee containers.